src/classes/GiftBatchStatusGroup.cls

```diff
@@ -0,0 +1,28 @@
+public inherited sharing class GiftBatchStatusGroup {
+    final private String status;
+    final private String paymentStatus;
+    private Integer recordCount;
+
+    public GiftBatchStatusGroup(AggregateResultProxy result) {
+        Map<String,Object> resultProxy = result.getValuesByAlias();
+        paymentStatus = (String) resultProxy.get(Schema.SObjectType.DataImport__c.fields.Payment_Status__c.getName());
+        status = (String) resultProxy.get(Schema.SObjectType.DataImport__c.fields.Status__c.getName());
+        recordCount = (Integer) resultProxy.get('recordCount');
+    }
+
+    public Integer recordCount() {
+        return this.recordCount;
+    }
+
+    public Boolean isImported() {
+        return status == 'Imported';
+    }
+
+    public Boolean isProcessingFailed() {
+        return status == 'Failed';
+    }
+
+    public Boolean isPaymentFailed() {
+        return paymentStatus == 'DECLINED';
```

Comment:
We sure that this is exactly what the payment status will be after a failed credit card transaction events are processed?

Reply:
Yes, according to this [quip doc](https://salesforce.quip.com/771RADnOwEsG#FSHACAgtkdh).

But I do wish there was a more definitive way to get all the possible status values. Perhaps we should just establish our own internal enum at the very least that's used across all classes needing to reference payment status.

Reply:
scratch that, we do have one stashed inside `GE_PaymentServices`

Reply:
as a side note in the BGE CC work I've pulled the enum out to a top level instead of having it as an inner class https://github.com/SalesforceFoundation/NPSP/commit/4187846676d50d9effc76d07f893177bf443f20a